Documentation: updated slice and join documentation (#7199)
opencv/cvat: site/content/en/docs/manual/advanced/slice-and-join.md

---
title: 'CVAT joining and slicing labels'
linkTitle: 'Joining and Slicing Labels'
weight: 18
description: 'This section explains how to slice or join several labels'
---

In CVAT you can modify labels by either joining multiple labels into
a single label or slicing a single label into several labels.

This document provides guidance on how to perform these operations effectively.

See:

- [Joining CVAT labels](#joining-cvat-labels)
- [Slicing CVAT labels](#slicing-cvat-labels)

## Joining CVAT labels

The **Join masks** tool (![Join masks tool icon](/images/join-masks-icon.jpg)),
is specifically designed to work with mask annotations.

This tool is useful in scenarios where a single object
in an image is annotated with multiple labels,
and there is a need to merge these labels into a single, unified label.

![Join masks](/images/joining-tool-01.jpg)

To join masks, do the following:

1. From the [**Edit**](/docs/manual/basics/controls-sidebar/#edit) block,
   select **Join masks** ![Join masks tool icon](/images/join-masks-icon.jpg).
2. Click on the labels, to select masks that you want to join.

> 3. (Optional) To remove selection double click on the mask.

Reviewer: Why double click?
If you double click you will remove selection and select again

Author: Weird... I've updated it but changes are not there....

> 3. (Optional) To remove selection double click on the mask.

Reviewer: Click one more time. 
Double click is another mouse event.

4. Click again on **Join masks** ![Join masks tool icon](/images/join-masks-icon.jpg)
  (**J**) to execute the join operation.

> Upon completion, the selected masks will be merged into a single mask.

Reviewer: ```suggestion
Upon completion, the selected masks will be joined into a single mask.
```

![Join masks gif](/images/joining-tool-02.gif)

## Slicing CVAT labels

The **Slice mask/polygon** (![Slicing tool icon](/images/slicing-tool-icon.jpg))
is compatible with both mask and polygon annotations.

> This tool is useful in scenarios where multiple object in an image
> are annotated with one label labels,

Reviewer: ```suggestion
This tool is useful in scenarios where multiple objects in an image
are annotated with one shape,
```

> and there is a need to slice this label into multiple labels.

Reviewer: ```suggestion
and there is a need to slice this shape into multiple parts.
```

![Slicing tool](/images/slicing-tool-01.jpg)

To slice mask or polygon, do the following:

1. From the [**Edit**](/docs/manual/basics/controls-sidebar/#edit) block,
   select **Slice mask/polygon** ![Slicing tool icon](/images/slicing-tool-icon.jpg).
2. Click on the label you intend to slice.
   A black contour will appear around the selected label.
3. Set an initial point for slicing by clicking on the contour.
4. Draw a line across the label to define the slicing path.
   <br> **Note**: The line cannot cross itself.

>    <br> **Note**: The line cannot cross the contour more than twice.

Reviewer: Let's mention right click to cancel the latest point and holding Shift to add points automatically on cursor movement

---
title: 'Join and slice tools'
linkTitle: 'Join and slice tools'
weight: 18
description: 'This section explains how to slice or join several labels'
---

In CVAT you can modify shapes by either joining multiple shapes into
a single label or slicing a single label into several shapes.

This document provides guidance on how to perform these operations effectively.

See:

- [Joining CVAT labels](#joining-cvat-labels)
- [Slicing CVAT labels](#slicing-cvat-labels)

> ## Joining CVAT labels

Reviewer: ```suggestion
## Joining masks
```

Obviously we speak about CVAT, so, I do not really see a reason to add `CVAT`

Author: it's for google search

The **Join masks** tool (![Join masks tool icon](/images/join-masks-icon.jpg)),
is specifically designed to work with mask annotations.

This tool is useful in scenarios where a single object
in an image is annotated with multiple shapes,

> and there is a need to merge these shapes into a single, unified shape.

Reviewer: ```suggestion
and there is a need to merge these shapes into a single one.
```

![Join masks](/images/joining-tool-01.jpg)

To join masks, do the following:

1. From the [**Edit**](/docs/manual/basics/controls-sidebar/#edit) block,
   select **Join masks** ![Join masks tool icon](/images/join-masks-icon.jpg).

> 2. Click on the labels, to select masks that you want to join.

Reviewer: ```suggestion
2. Click on the canvas area, to select masks that you want to join.
```

> 3. (Optional) To remove the selection click one more time on the mask.

Reviewer: ```suggestion
3. (Optional) To remove the selection click the mask one more time.
```

4. Click again on **Join masks**![Join masks tool icon](/images/join-masks-icon.jpg)
   (**J**) to execute the join operation.

Upon completion, the selected masks will be joined into a single mask.

![Join masks gif](/images/joining-tool-02.gif)

> ## Slicing CVAT labels

Reviewer: ```suggestion
## Slicing polygons and masks
```

The **Slice mask/polygon** (![Slicing tool icon](/images/slicing-tool-icon.jpg))
is compatible with both mask and polygon annotations.

This tool is useful in scenarios where multiple objects in an image
are annotated with one shape,

> and there is a need to slice this shape into multiple parts.

Reviewer: Maybe we should add, that a shape can be sliced only in two parts in time.
Apply the slice tool several times to split a shape to as many parts as you need.

![Slicing tool](/images/slicing-tool-01.jpg)

To slice mask or polygon, do the following:

1. From the [**Edit**](/docs/manual/basics/controls-sidebar/#edit) block,
   select **Slice mask/polygon** ![Slicing tool icon](/images/slicing-tool-icon.jpg).

> 2. Click on the label you intend to slice.
>    A black contour will appear around the selected label.

Reviewer: ```suggestion
2. Click on the shape you intend to slice.
   A black contour will appear around the selected shape.
```

3. Set an initial point for slicing by clicking on the contour.

> 4. Draw a line across the label to define the slicing path.

Reviewer: ```suggestion
4. Draw a line across the shape to define the slicing path.
```
